packages/bug-detectors/internal/prototype-pollution.ts

```diff
@@ -0,0 +1,565 @@
+} from "@jazzer.js/hooking";
+
+import { bugDetectorConfigurations } from "../configuration";
+
+// Allow the user to configure this bug detector in the custom-hooks file (if any).
+class PrototypePollutionConfig {
+	private excludedExactMatches: string[] = [];
+	private _instrument = false;
+
+	/**
+	 * Excludes one specific value of the `__proto__` property from being reported as a Prototype Pollution finding.
+	 * This is only relevant when instrumenting assignment expressions and variable declarations.
+	 * @param protoValue - stringified value of the `__proto__` for which no finding should be reported.
+	 */
+	addExcludedExactMatch(protoValue: string): PrototypePollutionConfig {
+		this.excludedExactMatches.push(protoValue);
+		return this;
+	}
+
+	/**
+	 * Enables instrumentation of assignment expressions and variable declarations.
+	 * This is a costly operation that might find non-global Prototype Pollution.
+	 * However, it also might result in false positives. Use `addExcludedExactMatch` to exclude specific values from
+	 * being reported as Prototype Pollution findings.
+	 */
+	instrumentAssignmentsAndVariableDeclarations(): PrototypePollutionConfig {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		if (global.options.dryRun) {
+			console.error(
+				"[Prototype Pollution Configuration] The configuration option " +
+					"instrumentAssignmentsAndVariableDeclarations() is not supported in dry run mode.\n" +
+					"  Either disable dry-run mode or remove this option from custom hooks.\n" +
+					"  Jazzer.js initial arguments:"
+			);
+			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+			// @ts-ignore
+			console.error(global.options);
+			// We do not accept conflicting configuration options: abort.
+			process.exit(1);
+		}
+		this._instrument = true;
+		return this;
+	}
+
+	getExcludedExactMatches(): string[] {
+		return this.excludedExactMatches;
+	}
+
+	getInstrumentAssignmentsAndVariableDeclarations(): boolean {
+		return this._instrument;
+	}
+}
+
+const config: PrototypePollutionConfig = new PrototypePollutionConfig();
+
+// Add this bug detector's config to the global config map.
+bugDetectorConfigurations.set("prototype-pollution", config);
+
+interface PrototypePollution {
+	getProtoSnapshot: typeof getProtoSnapshot;
+	detectPrototypePollution: typeof detectPrototypePollution;
+	protoSnapshotsEqual: typeof protoSnapshotsEqual;
+}
+
+declare global {
+	// eslint-disable-next-line no-var
+	var PrototypePollution: PrototypePollution;
+}
+
+// Make these functions available to instrumentation plugins and the user via the global object.
+globalThis.PrototypePollution = {
+	getProtoSnapshot: getProtoSnapshot,
+	detectPrototypePollution: detectPrototypePollution,
+	protoSnapshotsEqual: protoSnapshotsEqual,
+};
+
+// Keep track of statements and expressions that should not be instrumented.
+// This is necessary to avoid infinite recursion when instrumenting code.
```

Comment:
Does this only prevent recursive instrumentation of code within this particular bug detector? Since it adds assignments, what were to happen if another instrumentation plugin also instruments assignments, potentially by adding additional assignments?